test/jdk/com/sun/crypto/provider/Mac/Test6205692.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @library /test/lib
+ * @bug 6205692
+ * @summary Verify MacSpi
+ */
+
+import jdk.test.lib.Utils;
+
+import javax.crypto.Mac;
+import javax.crypto.MacSpi;
+import java.nio.ByteBuffer;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.Provider;
+import java.security.spec.AlgorithmParameterSpec;
+
+public class Test6205692 {
+
+    public boolean execute() throws Exception {
+
+        ByteBuffer byteBuffer = null;
+
+        MyMac mm = new MyMac(new MyMacSpi(), null, null);
+
+        Utils.runAndCheckException(() -> mm.updateSpi(byteBuffer),
+                NullPointerException.class);
+
+        return true;
+    }
+
+    public static void main(String[] args) throws Exception {
+        Test6205692 test = new Test6205692();
+
+        if (test.execute()) {
+            System.out.println(test.getClass().getName() + ": passed!");
+        }
+    }
+
+    private static class MyMacSpi extends MacSpi {
+
+        /*
+         * This is the important part; the rest is blank mandatory overrides
+         */
+        public void engineUpdate(ByteBuffer input) {
+            super.engineUpdate(input);
+        }
+
+        @Override
+        protected int engineGetMacLength() {
+            return 0;
+        }
+
+        @Override
+        protected void engineInit(Key key, AlgorithmParameterSpec params) throws InvalidKeyException, InvalidAlgorithmParameterException {
+        }
+
+        @Override
+        protected void engineUpdate(byte input) {
+        }
+
+        @Override
+        protected void engineUpdate(byte[] input, int offset, int len) {
+        }
+
+        @Override
+        protected byte[] engineDoFinal() {
+            return new byte[0];
+        }
+
+        @Override
+        protected void engineReset() {
+        }
+    }
+
+    private static class MyMac extends Mac {
+        private MyMacSpi spi;
+
+        public MyMac(MyMacSpi macSpi, Provider provider, String algorithm) {
+            super(macSpi, provider, algorithm);
+            spi = macSpi;
+        }
+
+        public void updateSpi(ByteBuffer input) {
+            spi.engineUpdate(input);
+        }
+    }
+}
```

Comment:
Can remove all these and directly instantiate the dummy MacSpi object and call the public engineUpdate(ByteBuffer) method in the execute method.

Reply:
removed! please re-review